presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -147,52 +136,112 @@ public void testJoinDynamicFilteringSingleValue()
+                ImmutableSet.of(1, ORDERS_COUNT, PART_COUNT));
+    }
+
+    @Test
+    public void testSemiJoinDynamicFilteringNone()
+    {
+        // Probe-side is not scanned at all, due to dynamic filtering:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders WHERE orders.totalprice < 0)",
+                ImmutableMap.of(JOIN_DISTRIBUTION_TYPE, BROADCAST.name()),
+                0,
+                ImmutableSet.of(0, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testSemiJoinLargeBuildSideNoDynamicFiltering()
+    {
+        // Probe-side is fully scanned because the build-side is too large for dynamic filtering:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders)",
+                ImmutableMap.of(JOIN_DISTRIBUTION_TYPE, BROADCAST.name()),
+                toIntExact(LINEITEM_COUNT),
+                ImmutableSet.of(LINEITEM_COUNT, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testPartitionedSemiJoinNoDynamicFiltering()
+    {
+        // Probe-side is fully scanned, because local dynamic filtering does not work for partitioned joins:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders WHERE orders.totalprice < 0)",
+                ImmutableMap.of(JOIN_DISTRIBUTION_TYPE, PARTITIONED.name()),
+                0,
+                ImmutableSet.of(LINEITEM_COUNT, ORDERS_COUNT));
+    }
+
+    @Test
+    public void testSemiJoinDynamicFilteringSingleValue()
+    {
+        assertQueryResult("SELECT orderkey FROM orders WHERE comment = 'nstructions sleep furiously among '", 1L);
+        assertQueryResult("SELECT COUNT() FROM lineitem WHERE orderkey = 1", 6L);
+
+        assertQueryResult("SELECT partkey FROM part WHERE comment = 'onic deposits'", 1552L);
+        assertQueryResult("SELECT COUNT() FROM lineitem WHERE partkey = 1552", 39L);
+
+        // Join lineitem with a single row of orders
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem WHERE lineitem.orderkey IN (SELECT orders.orderkey FROM orders WHERE orders.comment = 'nstructions sleep furiously among ')",
+                ImmutableMap.of(JOIN_DISTRIBUTION_TYPE, BROADCAST.name()),
+                6,
+                ImmutableSet.of(6, ORDERS_COUNT));
+
+        // Join lineitem with a single row of part
+        assertDynamicFiltering(
+                "SELECT l.comment FROM lineitem l WHERE l.partkey IN (SELECT p.partkey FROM part p WHERE p.comment = 'onic deposits')",
+                ImmutableMap.of(JOIN_DISTRIBUTION_TYPE, BROADCAST.name()),
+                39,
+                ImmutableSet.of(39, PART_COUNT));
+    }
+
+    private void assertDynamicFiltering(String selectQuery, Map<String, String> sessionProperties, int expectedRowCount, Set<Integer> expectedOperatorRowsRead)
```

Comment:
Would be better to have `Session session` as parameter instead of `Map<String, String> sessionProperties`, then we can use `withBroadcastJoin()` or `withPartitionedJoin()` as input

Reply:
Sounds good. Will update once @sopel39 makes more comments.